helpers.lua

```diff
@@ -260,35 +260,44 @@ function refresh_logged_in()
         end
     end
 
-    -- If client set the `Authorization` header before reaching the SSO,
-    -- we want to match user and password against the user database.
-    --
-    -- It allows to bypass the cookie-based procedure with a per-request
-    -- authentication. This is useful to authenticate on the SSO during
-    -- curl requests for example.
+    return false
+end
 
+-- If client set the `Authorization` header before reaching the SSO,
+-- we want to match user and password against the user database.
+--
+-- It allows to bypass the cookie-based procedure with a per-request
+-- authentication. This is useful to authenticate on the SSO during
+-- curl requests for example.
+function parse_auth_header()
     local auth_header = ngx.req.get_headers()["Authorization"]
 
     if auth_header then
         _, _, b64_cred = string.find(auth_header, "^Basic%s+(.+)$")
-        if b64_cred == nil then
-            is_logged_in = false
-            return is_logged_in
-        end
-        _, _, user, password = string.find(ngx.decode_base64(b64_cred), "^(.+):(.+)$")
-        user = authenticate(user, password)
-        if user then
-            logger.debug("User got authenticated through basic auth")
-            authUser = user
-            is_logged_in = true
-        else
-            is_logged_in = false
+        if b64_cred ~= nil then
+            _, _, user, password = string.find(ngx.decode_base64(b64_cred), "^(.+):(.+)$")
+            user = authenticate(user, password)
+            if user then
+                logger.debug("User got authenticated through basic auth")
+                is_logged_in = true
+                authUser = user
+
+                if has_access(permission, user) then
+                    refresh_user_cache(user)
+
+                    -- If Basic Authorization header are enable for this permission,
+                    -- add it to the response
+                    if permission["auth_header"] then
+                        set_headers(user)
+                    end
+
+                    return pass()
+                else
+                    return redirect(conf.portal_url)
+                end
```

Comment:
I guess it works ... but ideally it would be nice if we can keep this part of the logic in access.lua for the sake of clarity :/ Though I'm trying to see how we can do this and it feels like there's no easy / elegant way ...

Reply:
We can copy pasta the content for the function parse_auth_header in place we call it in access.lua? (As it was in versions < 4.1